torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -401,24 +403,25 @@ std::shared_ptr<rpc::FutureMessage> DistEngine::runEngineAndAccumulateGradients(
               DistAutogradContainer::getInstance().getWorkerId(),
               ": ",
               futureGrads->tryRetrieveErrorMessage());
-          accumulateGradFuture->setError(errorMsg);
+          accumulateGradFuture->setError(std::make_exception_ptr(
```

Comment:
any reason we cannot directly feed in the errMsg?

Reply:
ivalue::Future now only has one `setError` API which just consists of `exception_ptr`. The overall idea was that you usually always want to pass in an `exception_ptr` to it so it  can be appropriately rethrown. If you pass in an exception message from another exception (ex: e.what()) some information is lost.